File: Modulo 1/set1/mario-desafio/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int altura;

    // solicita altura válida (1-8)
    do
    {
        altura = get_int("Altura: ");
    }
    while (altura < 1 || altura > 8);

    // constroe as pirâmides
    for (int i = 0; i < altura; i++)
    {
        // espaços antes da primeira pirâmide
        for (int j = 0; j < altura - i - 1; j++)
        {
            printf(" ");
        }

        // primeira pirâmide
        for (int k = 0; k <= i; k++)
        {
            printf("#");
        }

        // espaço entre as pirâmides (sempre 2)
        printf("  ");

        // segunda pirâmide
        for (int l = 0; l <= i; l++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: Modulo 6/set6/mario-desafio/mario.py
from cs50 import get_int

def main():
    # Solicitar altura válida (1-8)
    while True:
        height = get_int("Height: ")
        if height is not None and 1 <= height <= 8:
            break

    # Imprimir as duas pirâmides
    for i in range(1, height + 1):
        # Espaços antes da primeira pirâmide
        print(" " * (height - i), end="")
        # Primeira pirâmide
        print("#" * i, end="")
        # Espaço entre as pirâmides (2 espaços)
        print("  ", end="")
        # Segunda pirâmide
        print("#" * i)

if __name__ == "__main__":
    main()
